libvast/vast/format/json/field_selector.hpp

```diff
@@ -23,23 +23,32 @@
 
 namespace vast::format::json {
 
-struct suricata {
-  suricata() {
+struct field_selector_specification {
+  const char* field;       ///< The field containing the type name.
+  const char* prefix;      ///< The stripped type name.
+  const char* reader_name; ///< The name of the json reader.
+};
+
+template <field_selector_specification (*FieldSelectorSpecification)()>
```

Comment:
```suggestion
template <field_selector_specification FieldSelectorSpecification>
```

Why not this?

Reply:
Cannot work until C++20, NTTPs are too restricted.

Reply:
And what speaks against `template <class Specification>`?

Reply:
That get's you harder to read compile time errors when you forget to update a specification.

Reply:
I changed it to some middle ground.